dashboard/app/helpers/curriculum_helper.rb

```diff
@@ -26,12 +26,20 @@ def validate_key_format
     return true
   end
 
-  # retrieves the matching UnitGroup or Unit associated with a course version and offering
+  # retrieves the course version matching the UnitGroup or Unit with a name of course_name
   def self.find_matching_course_version(course_name)
     matching_unit_group = UnitGroup.get_from_cache(course_name)
     return matching_unit_group.course_version if matching_unit_group
     matching_standalone_course = Unit.get_from_cache(course_name, raise_exceptions: false)
     return matching_standalone_course.course_version if matching_standalone_course&.is_course
     return nil
   end
+
+  # retrieves a unit or unit group matching course name
```

Comment:
I think I understand the necessity of this in addition to the function above because we won't necessarily know if it's a unit or unitgroup? But we know from the start if we want unit/unit group or the course version?

Reply:
Yes, pretty much. We don't know whether we want the content root of a course version (ie the unit group or standalone unit). This function will also find, say, `csd2-2022`, which the above function would not.

Definitely open to suggestions on how to improve the comments here!